Expensify/App
# Logic to support closed account receiving a link to set password and for unvalidated and closed accounts to get the email the moment they log into ecash

### Details
~On HOLD till I get some confirmation in the issue.~

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/163196

### Tests
1. Make sure you have the code from this [web-e](https://github.com/Expensify/Web-Expensify/pull/30904) and [auth](https://github.com/Expensify/Auth/pull/5552) PR and have done make etc.
2. I would recommend testing on a clean slate which makes this easier i.e., go to sql.sh and run `delete from notifications;`
3. Go to ecash, click on new chat, enter a new gmail account. Make a note of the email so you don't forgot. Log out.
4. In ecash log into the email that you entered in the previous step and click continue. It should be a closed account so in the network tab you should see reopenAccount request like so,
![image](https://user-images.githubusercontent.com/14356145/117718967-b0d42700-b191-11eb-8d97-276a8187bbc7.png)
5. Run `select * from notifications` and confirm you see a "SetCashPassword" template notification. 
6. On dev, notification emails won't get send to gmail domains so you'll need to run 
```
update notifications set json =  json_set(json(notifications.json), '$.to', '<your_expensify_email>');
```
6. Edit this [line](https://github.com/Expensify/Web-Expensify/blob/92e974b63a5f5121afe1f44173219aaf57f5b1f2/lib/Notification/Standard.php#L688) and replace `$this->parameters['to']` with your gmail test account otherwise the validate code won't work 
7. In vssh run `php /vagrant/Web-Expensify/script/notifyall.php`.
8. Check your email. Click on the link. 
9. Confirm you are redirected to a setpassword screen like so,
![image](https://user-images.githubusercontent.com/14356145/117517936-a0823900-af52-11eb-80d0-c0686687e464.png)
10. Enter a password and confirm it works just fine.

### QA Steps
None. Its a little involved with multiple flows and cannot be tested on staging so once this is live I'll ask tom to test.

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
